.github/workflows/test-and-publish-nightly.yml

```diff
@@ -4,25 +4,92 @@ on:  # We are very liberal in terms of triggering builds. This should be revisit
   - push
   - pull_request
 
+env:
+  java_default_distribution: corretto
+  java_default_version: 11
+
 jobs:
   build-jars:
     runs-on: ubuntu-latest
+    name: Build JARs
     steps:
       - name: Checkout sources
         uses: actions/checkout@v4
-      - name: Build jars
+      - name: Build JARs
         run: make jar
-      - name: Upload async-profiler.jar
+      - name: Upload JARs
         uses: actions/upload-artifact@v4
         with:
-          name: async-profiler.jar
-          path: build/jar/async-profiler.jar
-      - name: Upload jfr-converter.jar
+          name: async-profiler-jars
+          path: build/jar/*
+          if-no-files-found: error
+  build-and-upload-binaries:
+    strategy:
+      matrix:
+        include:
+          - runson:
+              display: linux-arm64
+              name: ubuntu-24.04-arm
+            container: "public.ecr.aws/async-profiler/asprof-builder-arm:latest"
+          - runson:
+              display: linux-x64
+              name: ubuntu-latest
+            container: public.ecr.aws/async-profiler/asprof-builder-x86:latest
+          - runson:
+              display: macos-arm64
+              name: macos-14
+          - runson:
+              display: macos-x64
```

Comment:
Why do we build on macos-x64?
`macos-arm64` produces universal (aka "fat") binaries that can run both on ARM64 and x64.

Reply:
6fb94f7